### PR TITLE
Force to marshall nill types

### PR DESCRIPTION
**Pull request by acazau**

### CODE REVIEW

File: parser/testdata/complexType.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+	tt := time.Time(*t.Time)
+	if tt.IsZero() {
+		return e.EncodeElement("", s)
+	}
+
+	return e.EncodeElement(t, s)
+}
+
+type myelements struct {
```
Comment (snedzad) [inline parser/testdata/complexType.go:58]: `myElements`?

File: parser/testdata/complexType.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	Nonboth     *string        `xml:"nonboth"`
+	Minzero     *int32         `xml:"minzero"`
+	Nilint      intReqNil      `xml:"nilint"`
+	Nilstring   stringReqNil   `xml:"nilstring"`
+	Minzeronil  *string        `xml:"minzeronil"`
+	DateOfBirth dateTimeReqNil `xml:"DateOfBirth"`
+}
```
Comment (snedzad) [inline parser/testdata/complexType.go:65]: One of the team's earlier discussion laid out the code organization in `.go` files. The conclusion was to order elements as following:
// consts
// vars
// types
// funcs

File: parser/testdata/complexType.go (hunk 1 above)
Comment (acazau) [inline parser/testdata/complexType.go:58]: this file is generated by the tool, this is a test result for a particular input
"myElements" is defined on the wsdl file as the name of the element

Comment (snedzad) [inline parser/testdata/complexType.go:58]: Oh, didn't realize that. Sorry, will merge as it is.

File: parser/testdata/complexType.go (hunk 2 above)
Comment (acazau) [inline parser/testdata/complexType.go:65]: we should continue this discussions, methods should be close to the type they act on, for easy discoverability

**Pull request merged**